realsense_camera/src/base_nodelet.cpp

```diff
@@ -724,81 +726,94 @@ namespace realsense_camera
   /*
    * Copy frame data from realsense to member cv images.
    */
-  void BaseNodelet::getStreamData(rs_stream stream_index)
+  void BaseNodelet::getStreamData(rs_stream stream_index, rs::frame & frame)
   {
     if (stream_index == RS_STREAM_DEPTH)
     {
       // fill depth buffer
-      image_depth16_ = reinterpret_cast<const uint16_t *>(rs_get_frame_data(rs_device_, stream_index, 0));
-      if (depth_scale_meters_ == MILLIMETER_METERS)
+      image_depth16_ = reinterpret_cast<const uint16_t *>(frame.get_data());
+      float depth_scale_meters = rs_get_device_depth_scale(rs_device_, &rs_error_);
```

Comment:
Should this function be called in here? getStreamData is called the most frequently in the code.
Can we initialize this something else?  Or make it a local static and initialize it once.

At a minimum we need to add a check error for the rs_error_